pom.xml

```diff
@@ -95,6 +95,7 @@
     <maven.checkstyle.plugin.version>2.17</maven.checkstyle.plugin.version>
     <maven.build-helper.plugin.version>1.12</maven.build-helper.plugin.version>
     <maven.eclipse.plugin.version>2.10</maven.eclipse.plugin.version>
+    <maven.shade.plugin.version>3.4.1</maven.shade.plugin.version>
```

Comment:
@cnauroth if this wasn't an explicit dependency prior, where was hive pulling in this maven.shade.plugin from? Any idea?

Reply:
Hello @nrg4878 . Thank you for the reply.

The Hive root pom.xml specifies the common Apache pom as its parent:

```
    <parent>
      <groupId>org.apache</groupId>
      <artifactId>apache</artifactId>
      <version>23</version>
    </parent>
```

From the output of `mvn help:effective-pom`, I can see that pom specifies maven-shade-plugin version 3.1.1, so for the maven-shade-plugin invocations in Hive, this is the version getting picked up:

https://github.com/apache/maven-apache-parent/blob/apache-23/pom.xml#L308

There is a more recent version 27 release of the Apache parent pom, which pins to maven-shade-plugin 3.3.0:

https://github.com/apache/maven-apache-parent/blob/apache-27/pom.xml#L301

I'm not sure if that version is also sufficient for the bugs I'm trying to address. I didn't take the approach of upgrading the parent pom, because it's potentially a larger change impacting more than just the maven-shade-plugin fix I was trying to target.